### PR TITLE
Improved logging and fixed issue with nullable fields for BQ Pushdown

### PR DESCRIPTION
**Pull request by fernst**

### CODE REVIEW

File: src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/util/BigQuerySQLEngineUtils.java [hunk 1]
```diff
@@ -87,6 +92,7 @@ public static Long getNumRows(BigQuery bigQuery, String project, String dataset,
                                                  table, dataset, project));
     }
 
+    LOG.info("Table {} contains {} records", table, bgTable.getNumRows().longValue());
```
Comment (tivv) [inline src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/util/BigQuerySQLEngineUtils.java]: Please surround with `if (LOG.isInfoEnabled())` `getNumRows` is definitely not free

**Review by tivv — APPROVED** [review]
LGTM with minor comment

File: src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/BigQuerySQLEngine.java [hunk 2]
```diff
@@ -218,18 +224,24 @@ public SQLDataset join(SQLJoinRequest sqlJoinRequest) throws SQLEngineException
...
     datasets.put(sqlJoinRequest.getDatasetName(), joinDataset);
 
     return joinDataset;
   }
 
   @Override
   public void cleanup(String datasetName) throws SQLEngineException {
+
     BigQuerySQLDataset bqDataset = datasets.get(datasetName);
+    //if (bqDataset == null || config.skipDelete()) {
```
Comment (albertshau) [inline src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/BigQuerySQLEngine.java]: unintended change?

File: src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/util/BigQuerySQLEngineUtils.java [hunk 3]
```diff
@@ -79,6 +83,7 @@ public static String getNewTableName(String runId) {
    * @return number of rows for this table.
    */
   public static Long getNumRows(BigQuery bigQuery, String project, String dataset, String table) {
+    LOG.info("Getting number of records stored in table {}", table);
```
Comment (albertshau) [inline src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/util/BigQuerySQLEngineUtils.java]: this seems more like a debug message to me

File: src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/util/BigQuerySQLEngineUtils.java (hunk 1 above)
Comment (albertshau) [reply src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/util/BigQuerySQLEngineUtils.java]: should set bgTable.getNumRows().longValue() in a variable and refer to it here in the log statement before returning it below.

File: src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/BigQuerySQLEngine.java (hunk 2 above)
Comment (fernst) [reply src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/BigQuerySQLEngine.java]: Yes! This is something I plan to introduce later as a way to add a few debugging behaviors to the plugin.

File: src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/util/BigQuerySQLEngineUtils.java (hunk 1 above)
Comment (fernst) [reply src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/util/BigQuerySQLEngineUtils.java]: The expensive operation here is the `Table bgTable = bigQuery.getTable(tableId)`. In that response, the number of rows is returned.

I will make this a local variable as I'm using this value in 2 places.

**Pull request merged**